### PR TITLE
docs: Add `/Formula/*` sharding

### PR DESCRIPTION
**Pull request by osalbahr**
Fixed all of `grep -rE '/Formula/.[^/]'` hyperlinks and a file path.

I think the same needs to be done for Cask

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

```console
% brew typecheck
Fetching gem metadata from https://rubygems.org/.......
Fetching sorbet 0.5.10461
Fetching yard-sorbet 0.6.1
Installing yard-sorbet 0.6.1
Installing sorbet 0.5.10461
Bundler::GemNotFound: Could not find commander-4.6.0.gem for installation
 
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/bundler-2.4.18/lib/bundler/source/rubygems.rb:161:in
`install'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/bundler-2.4.18/lib/bundler/installer/gem_installer.rb:54:in
`install'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/bundler-2.4.18/lib/bundler/installer/gem_installer.rb:16:in
`install_from_spec'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/bundler-2.4.18/lib/bundler/installer/parallel_installer.rb:156:in
`do_install'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/bundler-2.4.18/lib/bundler/installer/parallel_installer.rb:147:in
`block in worker_pool'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/bundler-2.4.18/lib/bundler/worker.rb:62:in
`apply_func'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/bundler-2.4.18/lib/bundler/worker.rb:57:in
`block in process_queue'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/bundler-2.4.18/lib/bundler/worker.rb:54:in
`loop'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/bundler-2.4.18/lib/bundler/worker.rb:54:in
`process_queue'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/bundler-2.4.18/lib/bundler/worker.rb:90:in
`block (2 levels) in create_threads'

An error occurred while installing commander (4.6.0), and Bundler
cannot continue.

In Gemfile:
  parlour was resolved to 8.1.0, which depends on
    commander
Error: failed to run `/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/bin/bundle install`!
```
-----

### CODE REVIEW

File: docs/Homebrew-homebrew-core-Maintainer-Guide.md [hunk 1]
```diff
@@ -95,7 +95,7 @@ The following checklist is intended to help maintainers decide on whether to mer
 
 - previously opened active PRs, as we would like to be fair to contributors who came first
 - patches/`inreplace` that have been applied to upstream and can be removed
-- comments in formula around `url`, as we do skip some versions (for example [`vim`](https://github.com/Homebrew/homebrew-core/blob/359dbb190bb3776c4d6a1f603a271dd8f186f077/Formula/vim.rb#L4) or [`v8`](https://github.com/Homebrew/homebrew-core/blob/359dbb190bb3776c4d6a1f603a271dd8f186f077/Formula/v8.rb#L4))
+- comments in formula around `url`, as we do skip some versions (for example [`vim`](https://github.com/Homebrew/homebrew-core/blob/359dbb190bb3776c4d6a1f603a271dd8f186f077/Formula/v/vim.rb#L4) or [`v8`](https://github.com/Homebrew/homebrew-core/blob/359dbb190bb3776c4d6a1f603a271dd8f186f077/Formula/v/v8.rb#L4))
```
Comment (Bo98) [inline docs/Homebrew-homebrew-core-Maintainer-Guide.md:98]: Same here

File: docs/Node-for-Formula-Authors.md [hunk 2]
```diff
@@ -48,8 +48,8 @@ Node modules should be installed to `libexec`. This prevents the Node modules fr
...
+* formulae for standard Node modules compatible with npm's global module format which should use [`std_npm_install_args`](#installing-global-style-modules-with-std_npm_install_args-to-libexec) (like [`apollo-cli`](https://github.com/Homebrew/homebrew-core/blob/790c4298124d49823e5b9286d07519c1c1a9f773/Formula/a/apollo-cli.rb) or [`webpack`](https://github.com/Homebrew/homebrew-core/blob/6bb687199709b9cfed362efc9581e0ef24917acd/Formula/w/webpack.rb))
+* formulae where the `npm install` call is not the only required install step (e.g. need to also compile non-JavaScript sources) which have to use [`local_npm_install_args`](#installing-module-dependencies-locally-with-local_npm_install_args) (like [`emscripten`](https://github.com/Homebrew/homebrew-core/blob/140d42bd9ada53cfd921abf964796dbf5bac1c3b/Formula/e/emscripten.rb) or [`grunt-cli`](https://github.com/Homebrew/homebrew-core/blob/0ab5d8e18f1dea2c4c57407a17f4aa4b8e985d76/Formula/g/grunt-cli.rb))
```
Comment (Bo98) [inline docs/Node-for-Formula-Authors.md:52]: Same here

File: docs/Reproducible-Builds.md [hunk 3]
```diff
@@ -17,7 +17,7 @@ def install
 end
 ```
 
-See the [`kustomize`](https://github.com/Homebrew/homebrew-core/blob/4c911aecde3c5a1c18df8e439a70c2f03b97297c/Formula/kustomize.rb#L32) formula for an example of using `time.iso8601` or the [`git-town`](https://github.com/Homebrew/homebrew-core/blob/4c911aecde3c5a1c18df8e439a70c2f03b97297c/Formula/git-town.rb#L24) formula for an example of using `time.strftime` with custom format specifiers.
+See the [`kustomize`](https://github.com/Homebrew/homebrew-core/blob/4c911aecde3c5a1c18df8e439a70c2f03b97297c/Formula/k/kustomize.rb#L32) formula for an example of using `time.iso8601` or the [`git-town`](https://github.com/Homebrew/homebrew-core/blob/4c911aecde3c5a1c18df8e439a70c2f03b97297c/Formula/g/git-town.rb#L24) formula for an example of using `time.strftime` with custom format specifiers.
```
Comment (Bo98) [inline docs/Reproducible-Builds.md:20]: Same here

File: docs/Reproducible-Builds.md [hunk 4]
```diff
@@ -39,7 +39,7 @@ def install
 end
 ```
 
-See the [`par` formula](https://github.com/Homebrew/homebrew-core/blob/09c6c09692561cfe49a04046fe2a769747d40494/Formula/par.rb#L30) for an example with a single file or the [`pari-elldata` formula](https://github.com/Homebrew/homebrew-core/blob/0d86ca42fa89759ba1ace07734778f09d18a5eb5/Formula/pari-elldata.rb#L28) for an example with multiple files.
+See the [`par` formula](https://github.com/Homebrew/homebrew-core/blob/09c6c09692561cfe49a04046fe2a769747d40494/Formula/p/par.rb#L30) for an example with a single file or the [`pari-elldata` formula](https://github.com/Homebrew/homebrew-core/blob/0d86ca42fa89759ba1ace07734778f09d18a5eb5/Formula/p/pari-elldata.rb#L28) for an example with multiple files.
```
Comment (Bo98) [inline docs/Reproducible-Builds.md:42]: Same here

File: docs/Formula-Cookbook.md [hunk 5]
```diff
@@ -136,7 +136,7 @@ We generally try not to duplicate system libraries and complicated tools in core
...
+Homebrew’s OpenSSL is [`keg_only`](https://rubydoc.brew.sh/Formula#keg_only-class_method) to avoid conflicting with the system so sometimes formulae need to have environment variables set or special configuration flags passed to locate our OpenSSL. You can see this mechanism in the [`bind`](https://github.com/Homebrew/homebrew-core/blob/024535144dfdbc107bcc76056361f9515289fe3e/Formula/b/bind.rb#L48) formula. Usually this is unnecessary because Homebrew sets up our [build environment](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/extend/ENV/super.rb) to favour finding [`keg_only`](https://rubydoc.brew.sh/Formula#keg_only-class_method) formulae first.
```
[the remark this answers is not in the document]
Comment (osalbahr) [reply docs/Formula-Cookbook.md:139]: Oh, I see. Yeah for permalinks I’ll revert back

**Comment by osalbahr** [comment]
Better?

File: docs/Formula-Cookbook.md [hunk 6]
```diff
@@ -869,7 +869,7 @@ if build.without? "ham"
 end
 ```
 
-[`option`](https://rubydoc.brew.sh/Formula#option-class_method) names should be prefixed with the words `with` or `without`. For example, an option to run a test suite should be named `--with-test` or `--with-check` rather than `--test`, and an option to enable a shared library `--with-shared` rather than `--shared` or `--enable-shared`. See the [alternative `ffmpeg`](https://github.com/homebrew-ffmpeg/homebrew-ffmpeg/blob/HEAD/Formula/ffmpeg.rb) formula for examples.
+[`option`](https://rubydoc.brew.sh/Formula#option-class_method) names should be prefixed with the words `with` or `without`. For example, an option to run a test suite should be named `--with-test` or `--with-check` rather than `--test`, and an option to enable a shared library `--with-shared` rather than `--shared` or `--enable-shared`. See the [alternative `ffmpeg`](https://github.com/homebrew-ffmpeg/homebrew-ffmpeg/blob/HEAD/Formula/f/ffmpeg.rb) formula for examples.
```
Comment (Bo98) [inline docs/Formula-Cookbook.md:872]: This is a third-party tap link where sharding does not apply.

**Comment by zgracem** [comment]
I really should have looked for an open PR _before_ starting this myself, whoops 😅 

I don't want to clobber the work that's already been done here, but since I went through the results of `ag '/homebrew-(?:cask|core)/blob/\w+/(?:Formula|Casks)/[^/]+?\.rb' docs/*.md` and manually checked/updated nearly all of the commit hashes and line-number anchors, I'd also hate for someone to needlessly duplicate all of _that_ work.

https://github.com/Homebrew/brew/compare/master...zgracem:brew:update-docs-urls

If there's anything I can do to make my changes easier to merge into this PR, please let me know!

**Comment by MikeMcQuaid** [comment]
@zgracem may be just worth opening your own PR and we can merge whichever is ready first!

**Comment by zgracem** [comment]
@MikeMcQuaid Sure! I've opened #15904 with my changes.

**Comment by osalbahr** [comment]
Superseded by the following PR:
- #15904

**Pull request closed without merging**